Troubleshooting.md

```diff
@@ -65,43 +76,41 @@ There are two possible reasons for this error.
 
-## The DEBUG CONSOLE throws Error: Could not find or load main class xyz
+## Cannot find a class with the main method
 ### Reason:
-You configure the incorrect main class name in `mainClass` of *launch.json*.
+When the `mainClass` is unconfigured in the *launch.json*, the debugger will resolve a class with main method automatically. This error indicates the debugger doesn't find any main class in the whole workspace.
 
 ### Try:
-1. Check whether the class name specified in `mainClass` exists and is in the right form.
-2. If the problem persists, it's probably because the language server doesn't load your project correctly. Please reference the [language server troubleshooting](#try) paragraph for more troubleshooting info.
+1. Check at least one main class exists in your workspace.
+2. If no main class exists, please create a main class first. Otherwise, it's probably because the language server fails to start. Please reference the [language server troubleshooting](#try) paragraph for more troubleshooting info.
 
-## The DEBUG CONSOLE throws ClassNotFoundException
+## No delegateCommandHandler for vscode.java.startDebugSession when starting Debugger
 ### Reason:
-This error indicates your application attempts to reference some classes which are not found in the entire classpaths.
+Cause of error is for now unknown, but something caused vscode java support and debugger to not be configured correctly.
 
 ### Try:
-1. Check whether you configure the required libraries in the dependency settings file (e.g. *pom.xml*).
-2. Run VS Code command *"Java: Force Java compilation"* to force the language server to rebuild the current project.
-3. If the problem persists, it's probably because the language server doesn't load your project correctly. Please reference the [language server troubleshooting](#try) paragraph for more troubleshooting info.
+1. Restart VS Code and the issue should disappear 
+2. If it continues to error try restart again, and if still a problem open an issue at [vscode-java-debug](https://github.com/Microsoft/vscode-java-debug)
 
-## Cannot find a class with the main method
+## Failed to resolve classpath:
 ### Reason:
-When the `mainClass` is unconfigured in the *launch.json*, the debugger will resolve a class with main method automatically. This error indicates the debugger doesn't find any main class in the whole workspace.
-
-### Try:
-1. Check at least one main class exists in your workspace.
-2. If no main class exists, please create a main class first. Otherwise, it's probably because the language server fails to start. Please reference the [language server troubleshooting](#try) paragraph for more troubleshooting info.
+Below are the common failure reasons.
+- 'C:\demo\org\microsoft\app\Main.java' is not a valid class name.
+- Main class 'org.microsoft.app.Main' doesn't exist in the workspace.
+- Main class 'org.microsoft.app.Main' isn't unique in the workspace.
+- The project 'demo' is not a valid java project.

```

Comment:
Is there anyway we can let the users know what is a valid java project?

Reply:
yes, the debugger will provide fix proposals quick pick for user to correct it if it finds the project is not valid.

Reply:
Upper case the word `Java`.